src/realm/sync/noinst/client_history_impl.cpp

```diff
@@ -395,39 +386,45 @@ void ClientHistory::integrate_server_changesets(const SyncProgress& progress,
 {
     REALM_ASSERT(num_changesets != 0);
 
+    std::uint_fast64_t downloaded_bytes_in_message = 0;
+    std::vector<Changeset> changesets;
+    changesets.resize(num_changesets); // Throws
+
+    try {
+        for (std::size_t i = 0; i < num_changesets; ++i) {
+            const RemoteChangeset& changeset = incoming_changesets[i];
+            downloaded_bytes_in_message += changeset.original_changeset_size;
+            parse_remote_changeset(changeset, changesets[i]); // Throws
+            changesets[i].transform_sequence = i;
+        }
+    }
+    catch (BadChangesetError& e) {
```

Comment:
It looks like `parse_remote_changeset` throws `TransformError` not `BadChangesetError`?

Reply:
Huh, that's sort of odd. Also it's concerning that there's apparently no tests which have the server sending bad changets to the client? I'll look into improving the test coverage here...

Reply:
I've added a test which would have caught this.